FindLib/etc/makefile

```diff
@@ -9,13 +10,27 @@ DEP_DIR := depends
 OPTIMIZER		=	-O0
 
 CC		     	=	gcc
-CFLAGS			=	$(OPTIMIZER) -Wall -Wno-multichar -Wno-ctor-dtor-privacy
+
+# This is needed to be able to compile on either nightlies, or beta4.
+# See https://dev.haiku-os.org/ticket/11818 (fixed for good in hrev57497).
+# We can simplify this once the haikuports buildmasters switch to beta5.
+
+HAIKU_HREV_NUM = $(shell uname -v | cut -d " " -f 1 | cut -d "+" -f 1 | sed s/hrev//)
+HAS_FIXED_FEATURES_H = $(shell if [ $(strip $(HAIKU_HREV_NUM)) -ge 57497 ]; then echo "YES"; else echo "NO"; fi)
+
+ifeq ($(strip $(HAS_FIXED_FEATURES_H)), YES)
+	CFLAGS	=	$(OPTIMIZER) -Wall -Wno-multichar -Wno-ctor-dtor-privacy -D_DEFAULT_SOURCE -DSTDC_HEADERS
```

Comment:
Why is _DEFAULT_SOURCE needed on nightlies?  Better define for instance -std=gnu++98

Reply:
I just copied what was added to main makefile in #4. There is a lengthy discussion there already.

Reply:
I understand, I mean everything should work out of the box with nightlies. If _DEFAULT_SOURCE is still needed with -std=gnu11++, then something is wrong...

Reply:
I haven't tried using `-std=gnu++98` instead, but I can try it.

Reply:
done. It works without the extra flags on nightly. (tested on x86_64)

I don't have a nightly x86_gcc2 installation, so I couldn't test gcc2.